src/core/control/xojfile/LoadHandler.cpp

```diff
@@ -950,7 +950,8 @@ void LoadHandler::fixNullPressureValues() {
 
     std::vector<std::vector<Point>> strokePortions;
     while (nextPositive != pressureBuffer.end()) {
-        auto nextNonPositive = std::find_if(nextPositive, pressureBuffer.end(), [](double v) { return v <= 0; });
+        auto nextNonPositive =
+                std::find_if(nextPositive, pressureBuffer.end(), [](double v) { return v <= 0 || v != v; });
```

Comment:
I'd make the variable name more explicit, and use std::isnan rather than v != v.

```suggestion
        auto nextInvalid =
                std::find_if(nextPositive, pressureBuffer.end(), [](double v) { return v <= 0 || std::isnan(v); });
```